### PR TITLE
mapbox map integration

### LINKED ISSUE #392: Consider replacing Apple maps with Mapbox maps in the publisher example app
Since AAT SDK uses Mapbox Nav SDK which relies on Mapbox Maps SDK, we could benefit from using the Mapbox maps SDK to see how it works together. We did that on Android and it allowed us to detect issues before the SDK is released. Therefore, it might be a good idea to do the same on iOS example apps.

### PR DESCRIPTION
**Pull request by arturjaworski**
Resolves #392

### CODE REVIEW

**Comment by lawrence-forooghian** [comment]
I'd like to take a look at this one but might not have a chance today. Please could you hold off on merging until I've taken a look? Thanks!

File: Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Settings/SettingsView.swift [hunk 1]
```diff
@@ -64,6 +64,9 @@ struct SettingsView: View {
                 Toggle(isOn: $viewModel.areRawLocationsEnabled) {
                     Text("Raw locations enabled")
                 }
+                Toggle(isOn: $viewModel.useMapBox) {
+                    Text("Use MapBox")
```
Comment (JakubJankowski) [inline Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Settings/SettingsView.swift:68]: This text kinda implies we can turn off the whole Mapbox integration, while we're just toggling which map will be used. So maybe `Use MapBox map` instead?

File: Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/ViewModel/MapViewModel.swift [hunk 2]
```diff
@@ -20,6 +20,9 @@ class MapViewModel: ObservableObject {
         }
     }
     private var updatedLocations: Set<CLLocation> = []
+    var useMapBox: Bool {
+        SettingsModel.shared.useMapBox
```
Comment (JakubJankowski) [inline Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/ViewModel/MapViewModel.swift:24]: Similar point to the above - maybe we should call vars like that `useMapBoxMap` instead?

File: Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Map/Map.swift [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    
+    var body: some View {
+        MapKit.Map(
+            coordinateRegion: region,
+            showsUserLocation: true
+        )
+    }
+}
+
+struct Map_Previews: PreviewProvider {
```
Comment (JakubJankowski) [inline Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Map/Map.swift:37]: This struct isn't used anywhere, can be removed?

File: Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Map/Map.swift [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    private var region: Binding<MKCoordinateRegion> {
+        Binding {
+            let span = span ?? MKCoordinateSpan(
+                latitudeDelta: 0.2,
+                longitudeDelta: 0.2
+            )
+            
+            return MKCoordinateRegion(center: center, span: span)
+        } set: { region in
+            DispatchQueue.main.async {
```
Comment (JakubJankowski) [inline Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Map/Map.swift:18]: Out of curiosity - do we need to do this setter async? Were there some UI stutters or something along those lines without it?

File: Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Map/Map.swift (hunk 3 above)
Comment (lawrence-forooghian) [reply Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Map/Map.swift:37]: @JakubJankowski This is [used by Xcode to display view previews](https://developer.apple.com/documentation/swiftui/previewprovider).

**Review by lawrence-forooghian — COMMENTED** [review]
Looks good, I agree with @JakubJankowski's comments re naming.

File: Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Settings/SettingsView.swift (hunk 1 above)
Comment (arturjaworski) [reply Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Settings/SettingsView.swift:68]: changed

File: Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/ViewModel/MapViewModel.swift (hunk 2 above)
Comment (arturjaworski) [reply Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/ViewModel/MapViewModel.swift:24]: changed

File: Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Map/Map.swift (hunk 4 above)
Comment (arturjaworski) [reply Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Map/Map.swift:18]: I've got some console message. Running in main thread resolved it.

File: Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -18,6 +18,9 @@
 		0CFCA307289131AB00A1A6B5 /* UserDefaults+Codable.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0CFCA305289131AB00A1A6B5 /* UserDefaults+Codable.swift */; };
 		0CFCA308289131AB00A1A6B5 /* CloseKeyboardHelper.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0CFCA306289131AB00A1A6B5 /* CloseKeyboardHelper.swift */; };
 		0CFCA30A2891325700A1A6B5 /* MainView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0CFCA3092891325700A1A6B5 /* MainView.swift */; };
+		21E6EBF328F836FF0057AC7C /* MapBoxMap.swift in Sources */ = {isa = PBXBuildFile; fileRef = 21E6EBF228F836FF0057AC7C /* MapBoxMap.swift */; };
```
[the remark this answers is not in the document]
Comment (arturjaworski) [reply Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI.xcodeproj/project.pbxproj:21]: changed

File: Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Map/Map.swift (hunk 3 above)
Comment (JakubJankowski) [reply Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/View/Map/Map.swift:37]: Ohh I see, disregard then!

**Review by JakubJankowski — APPROVED** [review]
LGTM 👍

**Pull request merged**